nifi-nar-bundles/nifi-riemann-bundle/nifi-riemann-processors/src/main/java/org/apache/nifi/processors/riemann/PutRiemann.java

```diff
@@ -0,0 +1,376 @@
+        successfulFlowFiles.add(flowFile);
+      } catch (NumberFormatException e) {
+        getLogger().warn(e.getMessage());
+        session.transfer(flowFile, REL_FAILURE);
+      }
+    }
+    try {
+      if (transport == Transport.TCP) {
+        Proto.Msg returnMessage = riemannClient.sendEvents(eventsQueue).deref(writeTimeout, TimeUnit.MILLISECONDS);
+        if (returnMessage == null) {
+          context.yield();
+          throw new ProcessException("Timed out writing to Riemann!");
+        }
+      } else {
+        riemannClient.sendEvents(eventsQueue);
+      }
+      riemannClient.flush();
+    } catch (Exception e) {
+      context.yield();
+      throw new ProcessException("Failed writing to Riemann\n" + e.getMessage());
+    }
+    session.transfer(successfulFlowFiles, REL_SUCCESS);
+    session.commit();
+  }
+
+  /**
+   * Converts a FlowFile into a Riemann Protobuf Event
+   */
+  private static class FlowFileToEvent {
+    protected static Event fromAttributes(ProcessContext context, List<PropertyDescriptor> customProperties,
+                                          FlowFile flowFile) {
+      Event.Builder builder = Event.newBuilder();
+
+      PropertyValue service = context.getProperty(ATTR_SERVICE).evaluateAttributeExpressions(flowFile);
+      if (service.getValue() != null && !service.getValue().equals("")) {
+        builder.setService(service.getValue());
+      }
+      PropertyValue description = context.getProperty(ATTR_DESCRIPTION).evaluateAttributeExpressions(flowFile);
+      if (description.getValue() != null && !description.getValue().equals("")) {
+        builder.setDescription(description.getValue());
+      }
+      PropertyValue metric = context.getProperty(ATTR_METRIC).evaluateAttributeExpressions(flowFile);
+      if (metric.getValue() != null && !metric.getValue().equals("")) {
+        builder.setMetricF(metric.asFloat());
+      }
+      PropertyValue time = context.getProperty(ATTR_TIME).evaluateAttributeExpressions(flowFile);
+      if (time.getValue() != null && !time.getValue().equals("")) {
+        builder.setTime(time.asLong());
+      }
+      PropertyValue state = context.getProperty(ATTR_STATE).evaluateAttributeExpressions(flowFile);
+      if (state.getValue() != null && !state.getValue().equals("")) {
+        builder.setState(state.getValue());
+      }
+      PropertyValue ttl = context.getProperty(ATTR_TTL).evaluateAttributeExpressions(flowFile);
+      if (ttl.getValue() != null && !ttl.getValue().equals("")) {
+        builder.setTtl(ttl.asFloat());
+      }
+      PropertyValue host = context.getProperty(ATTR_HOST).evaluateAttributeExpressions(flowFile);
+      if (host.getValue() != null && !host.getValue().equals("")) {
+        builder.setHost(host.getValue());
+      }
+      PropertyValue tags = context.getProperty(ATTR_TAGS).evaluateAttributeExpressions(flowFile);
+      if (tags.getValue() != null && !tags.getValue().equals("")) {
```

Comment:
Kind of general comment applying to all IF statements above. Since you are checking for empty strings, should/would values such as "  " be still considered an empty string? If so, should it be trimmed before checking for it being empty?